artichoke-backend/src/eval.rs

```diff
@@ -56,7 +57,13 @@ impl Eval for Artichoke {
         let context = Context::new(os_str_to_bytes(file.as_os_str())?.to_vec())
             .ok_or_else(|| ArgumentError::with_message("path name contains null byte"))?;
         self.push_context(context)?;
-        let code = self.read_source_file_contents(file)?.into_owned();
+        let code = self
+            .read_source_file_contents(file)
+            .map_err(|err| {
+                let message = format!("ruby: {} -- {}", err.message().as_bstr(), file.to_string_lossy());
```

Comment:
rather than using `to_string_lossy` here, can we construct the message as a `Vec<u8>`?

Reply:
Since [`Path` is not guarantee to be valid `UTF-8`](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_str), I'm not sure if we have many choices here. I added a test to demo the case when path contains invalid UTF-8: https://github.com/artichoke/artichoke/pull/2356/commits/19330239a7f1a17de10022abdcf4bc68d9065536#diff-f53af3ab96f4693c224425643c37f1dd017da0f569bff7ee5f747c04fa1a2e02R253

I'm not sure if I understand correctly, but displaying `Vec<u8>` will work but doesn't seem right https://github.com/choznerol/artichoke/commit/71c54d2c5852969c33becd952c7515f6fdfb769e

Using `.to_str()` will require `.unwrap()` and the newly added test case will panic: https://github.com/choznerol/artichoke/commit/5c2b973944ac386798c6fe31a425d83b416be4a0

Reply:
Update: found that `.to_string_loussy()` can be replaced by `.display()` and produce the same effect. Pushed cff55d2.